src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -123,21 +125,43 @@ void CALLBACK cfApiFetchDataCallback(const CF_CALLBACK_INFO *callbackInfo, const
         return;
     }
 
-    qint64 offset = 0;
+    // CFAPI expects sent blocks to be of a multiple of a block size.
+    // Only the last sent block is allowed to be of a different size than
+    // a multiple of a block size
+    constexpr auto cfapiBlockSize = 4096;
+    qint64 dataOffset = 0;
+    QByteArray protrudingData;
+
+    const auto alignAndSendData = [&](const QByteArray& receivedData) {
+        QByteArray data = protrudingData + receivedData;
+        protrudingData.clear();
+        if (data.size() < cfapiBlockSize) {
+            protrudingData = data;
+            return;
+        }
+        const auto protudingCount = data.size() % cfapiBlockSize;
```

Comment:
nitpick: s/protudingCount/protudingSize/ ?

I don't remember if I proposed "count" earlier but that makes me think in term of "number of blocks" while it's the size of a single block really.